### PR TITLE
Merge docker & podman tests

### PR DESCRIPTION
**Pull request by ilausuch**
In order to simplify maintenance we want to merge
both tests in one because there are a lot of
common source code lines.

- Related ticket: https://progress.opensuse.org/issues/101454
- Verification run:
- http://copland.arch.suse.de/tests/1080 (docker)
- http://copland.arch.suse.de/tests/1079 (podman)
- http://copland.arch.suse.de/tests/1087 (public cloud)
- http://openqa.suse.de/tests/8340025 (microos)
- http://openqa.suse.de/tests/8340473 (jeos) (running)

With the collaboration of @m-dati

### CODE REVIEW

File: tests/containers/container_runtime.pm [hunk 1]
```diff
@@ -27,25 +27,27 @@
 use Mojo::Base 'containers::basetest';
 use testapi;
 use utils;
+use registration;
```
Comment (pdostal) [inline tests/containers/container_runtime.pm:30]: Why is this here?

File: tests/containers/container_runtime.pm [hunk 2]
```diff
@@ -56,4 +58,15 @@ sub run {
...
 }
 
+sub post_fail_hook {
+    my ($self) = @_;
+    if ($self->{runtime} eq 'podman') {
+        select_console 'log-console';
+        script_run "podman version | tee /dev/$serialdev";
+        script_run "podman info --debug | tee /dev/$serialdev";
+        $self->SUPER::post_fail_hook;
+    }
```
Comment (pdostal) [inline tests/containers/container_runtime.pm:68]:
```suggestion
    }
    $self->SUPER::post_fail_hook;
```

**Review by pdostal — APPROVED** [review]

File: tests/containers/container_runtime.pm (hunk 2 above)
Comment (ilausuch) [reply tests/containers/container_runtime.pm:68]: Good catch. solved

File: tests/containers/container_runtime.pm (hunk 1 above)
Comment (ilausuch) [reply tests/containers/container_runtime.pm:30]: yes should be removed, was in the old podman.pm.
Solved

**Comment by jlausuch** [comment]
This will need a change too:
```
.//products/microos/main.pm:70:        loadtest 'containers/podman';
.//lib/main_publiccloud.pm:23:    loadtest 'containers/podman';
.//lib/main_publiccloud.pm:28:    loadtest 'containers/docker';
```

I haven't seen any podman|docker load test in yaml schedule. Almost all container tests in OSD and O3 are using `main_containers.pm`. We might need to improve that with `microos/main.pm` and `main_publiccloud.pm` in the future.

**Comment by jlausuch** [comment]
Looks good!

Maybe it would be a good time to rename "runtime" to "engine", because podman and docker are not actually "runtimes" per see as we have been calling them for long time. Some examples of runtimes are `containerd`, `crio`, etc..
What do you think?

**Comment by ilausuch** [comment]
I agree

**Comment by ilausuch** [comment]
renamed

**Review by jlausuch — APPROVED** [review]

**Review by mloviska — APPROVED** [review]
Me gusta!
